prow/plugins/trigger/generic-comment.go

```diff
@@ -70,6 +56,30 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 		return err
 	}
 
+	baseSHA, err := c.GitHubClient.GetRef(org, repo, "heads/"+pr.Base.Ref)
+	if err != nil {
+		return fmt.Errorf("failed to get baseSHA: %v", err)
+	}
+	presubmits, err := c.Config.GetPresubmits(c.GitClient, org+"/"+repo, baseSHA, pr.Head.SHA)
+	if err != nil {
+		return fmt.Errorf("failed to get presubmits: %v", err)
+	}
+
+	// Skip comments not germane to this plugin
```

Comment:
The `trigger` plugin is enabled widely (even on repos without any prowjobs configured) for convenience and avoids making API calls until it has determined the event is relevant to the plugin. Moving this down is going to be expensive in terms of API token usage. This will use 2 tokens before we've even checked if the body text includes a `/test` command or if the repo includes a `prow.yaml`.

Reply:
> What do you think about having a list of orgs and repos to enable prow.yaml files in the config? 

The whole thing is opt-in so the info is definitely available. We could do something along the lines of:

```
func getConfigForCommentEvent(c Client, org, repo string, number int) (
	presubmits []config.Presubmit,
	baseSHAGetter func() (string, error),
	prGetter func() (github.PullRequest, error),
	err error,
) {
	if !inRepoConfigEnabled {
		return c.Config.StaticPresubmits()[org+"/"+repo],
			func() (string, error) { return c.GitHubClient.GetRef(org, repo, "heads/"+pr.Base.Ref) },
			func() (github.PullRequest, error) { return c.GitHubClient.GetPullRequest(org, repo, number) },
			nil
	}

	pr, err := c.GitHubClient.GetPullRequest(org, repo, number)
	if err != nil {
		return nil, nil, nil, err
	}
	baseSHA, err := c.GitHubClient.GetRef(org, repo, "heads/"+pr.Base.Ref)
	if err != nil {
		return nil, nil, nil, fmt.Errorf("failed to get baseSHA: %v", err)
	}
	presubmits, err := c.Config.GetPresubmits(c.GitClient, org+"/"+repo, baseSHA, pr.Head.SHA)
	if err != nil {
		return nil, nil, nil, fmt.Errorf("failed to get presubmits: %v", err)
	}

	return presubmits,
		func() (string, error) { return baseSHA, nil },
		func() (github.PullRequest, error) { return pr, nil },
		nil
}
```

That way:

* We continue to use zero tokens until after ` Skip comments not germane to this plugin` check if in-repo config is disabled
* Can keep the exact same flow for both it being enabled and disabled
* Can be certain that we use the same baseSHA and headSHA for reading the `prow.yaml` and for the created ProwJobs.

WDYT?

Reply:
That sounds reasonable -- can we do it at the event demuxing level (so it's done once per event) and pass those closures to users where necessary through the plugin agent?